File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    void dft(TreeNode* root, int val, int& sum)
    {
        if (root == nullptr)
            return;
        
        val = val*10 + root->val;
        
        dft(root->left, val, sum);
        dft(root->right, val, sum);
        
        if (root->left == nullptr and root->right == nullptr)
            sum += val;
    }
public:
    int sumNumbers(TreeNode* root) {
        
        int sum = 0;
        
        dft(root, 0, sum);
        
        return sum;
    }
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
     int gSum = 0;
     void dfs(TreeNode root, int sum){
        if(root == null){return ;}
      sum = (sum*10) + root.val; 
        if(root.left == null && root.right == null) {
             gSum += sum;
               return ;
        }
        dfs(root.left,sum);  
        dfs(root.right,sum);
    return ; 
     }
    public int sumNumbers(TreeNode root) {
       dfs(root,0); 
      return gSum;  
    }
}